chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -40,13 +40,19 @@ use std::time::Instant;
 pub enum MainTransition {
     Genesis { chunk_extra: ChunkExtra, block_hash: CryptoHash, shard_id: ShardId },
     NewChunk(NewChunkData),
+    // TODO(#11881): this is temporary indicator that resharding happened in the
+    // state transition covered by state witness. Won't happen in production
+    // until resharding release. Must be removed and replaced with proper
+    // resharding state transition validation.
+    ShardLayoutChange,
```

Comment:
My thinking was that the resharding would be in addition to the existing MainTransition. I think we will first need to apply the chunk (if present) and only then validate resharding.

Reply:
I guess I mean that we can have both new chunk and resharding at the same time

Reply:
Is that what you mean by temporary? If so I'm happy to approve to unblock the progress, but still I'm curious about your thought of the proper structure for this logic.

Reply:
Yes, this is very temporary. It is just the easiest way to skip everything related to chunk validation in our case.